lib/libvarnish/vre.c

```diff
@@ -200,9 +203,24 @@ vre_match(const vre_t *code, const char *subject, size_t length, size_t offset,
 		AN(data);
 	}
 
-	matches =  pcre2_match(re, (PCRE2_SPTR)subject, length, offset,
+	matches = pcre2_match(re, (PCRE2_SPTR)subject, length, offset,
 	    options, data, code->re_ctx);
 
+	if (groups != NULL) {
+		AN(count);
+		AN(*count);
+		ovector = pcre2_get_ovector_pointer(data);
+		nov = pcre2_get_ovector_count(data);
```

Comment:
I am aware that I suggested using `pcre2_get_ovector_count()`, but later realized that we should also be able to use `matches`.

Reply:
That code was originally from `VRE_sub()`, and while they should be equivalent/off by one (because we create the match data based on the pattern) I decided not to for clarity: we see a clear relationship between `ovector` and `nov`.

Reply:
ack